### DISCUSSION TITLE
How to get ens name use Synchronized method

### DISCUSSION THREAD

**Discussion by aurevae** [wevm/wagmi#614] [Q&A]
I want to get the ens name as a Synchronized method. It seems as connectAsync method.

**Comment by tmm** [comment]
The query React Hooks (like `useEnsName`) only allow declarative fetching (`useConnect` is a mutation, not a query). If you want to imperatively fetch data, you could use actions from `@wagmi/core`:

```ts
import { fetchEnsName } from '@wagmi/core'

const ensName = await fetchEnsName({ address })
```

**Reply by aurevae** [reply]
Thank you. All i want is this
